src/main/java/io/vertx/codegen/CodeGen.java

```diff
@@ -214,19 +198,53 @@ private void processSerializerOrDeserializer(TypeElement converterElt, /*TypeEle
       mapper.setTargetType(tmf.create(paramType));
       mapper.setSelectors(selectors);
       mapper.setKind(MapperKind.STATIC_METHOD);
-      tmf.addDataObjectDeserializer(converterElt, returnType, mapper);
+      if (type == null) {
```

Comment:
In which cases, at this point, `type` should be null? When for example the target type is a bounded generic? If that's the case, and you want to handle it in this method, why in first instance do you pass `type` more than just reusing `returnType` and `paramType`?

Reply:
Previously the type might not be specified when using `@Mapper`, so it's a left over, now the code can be changed and type should never be null

Reply:
Maybe it's not a bad idea to allow having one serializer/deserializer to handle different subtypes? This kind of use-case is supported by this pr?

Reply:
I will also add tests for when the type is not correct, i.e if there is a mapper for Foo type and the converter param or return type does not extend nor is a super type of Foo

Reply:
no it is not supported, this would require actually code generators to perform casting with such mapper - so it would be another feature

Reply:
I explained myself badly: If the function is `Function<Fruit, JsonObject>`, could we use it as serializer of `Apple`? This should be trivial and doesn't require casting, right? For deserialization, the problem is quite different and it's not trivial to solve

Reply:
I don't know, that's a good point, it should theoretically work but it should not work currently because we are doing a check based on equality in the map, we don't crawl supper types.

It would also create more edge cases, e.g `Apple` extend `Fruit` and `RoundShape`, both having converters then it would be an issue.

Reply:
> It would also create more edge cases, e.g Apple extend Fruit and RoundShape, both having converters then it would be an issue.

Ah yeah that makes sense too... Can we have an issue for it and maybe think about if we want to address it and how later?

Reply:
I don't mind, but this feature would need to support bidi conversion if that's possible